addons/pos_coupon/static/src/js/OrderWidget.js

```diff
@@ -3,9 +3,15 @@ odoo.define('pos_coupon.OrderWidget', function (require) {
 
     const OrderWidget = require('point_of_sale.OrderWidget');
     const Registries = require('point_of_sale.Registries');
+    const { onWillRender } = owl;
 
     const PosCouponOrderWidget = (OrderWidget) =>
         class PosCouponOrderWidget extends OrderWidget {
+            setup() {
+                onWillRender(() => {
+                    Object.assign(this, this.getActiveProgramsProps());
+                });
+            }
```

Comment:
This is my personal opinion but it's better to initialize all the attributes in the `setup()` even if it's null and then refactor the `getActiveProgramsProps` to `updateActivePrograms` or something like that.
The reason behind this is because you are adding new attributes to the object by calling a method that modify the object. In term of readability, someone won't be able to get all the attributes of the object by looking at the constructor/setup and needs to go though other methods to see what attributes does the object have.

Reply:
![image](https://user-images.githubusercontent.com/92516038/154484185-02a7ad9f-24e6-4e26-aa16-414d9af8502c.png)
@trj-odoo  do you mean something like this?